#### This marks the beginning of generating a dataset sampled every minute.

In [1]:
import os
import pandas as pd
from dateutil import parser

#### Find all the CSV files with the same name collected from each date.


| **File Name**             | **Description**                                                                 |
|---------------------------|---------------------------------------------------------------------------------|
| **dataset.csv**            | All sensor data collected from PMS5003 Particulate Matter and Sensirion SCD30 Sensors |
| **no2dataset.csv**         | NO2 sensor data collected from ULPSM-NO2 968-047                                |
| **pm_aeroequal_data.csv**  | PM2.5 and PM10 sensor data collected from Aeroqual S500 Portable Air Quality Monitor |
| **no2_aeroequal_data.csv** | NO2 sensor data collected from Aeroqual S500 Portable Air Quality Monitor        |
| **iAnemometer.xls**        | Windspeed and Temperature collected from BTMETER BT-100 Anemometer               |


In [2]:
def find_csv_files(directory):
    pm_sensor_files = []
    no2_sensor_files = []
    pm_aeroequal_files = []
    no2_aeroequal_files = []
    anemometer_files = []
    
    for root, dirs, files in os.walk(directory):
        for file in files:
            if file.startswith("dataset.csv"):
                pm_sensor_files.append(os.path.join(root, file).replace("\\","/"))

            if file.startswith("no2dataset.csv"):
                no2_sensor_files.append(os.path.join(root, file).replace("\\","/"))

            if file.startswith("pm_aeroequal_data.csv"):
                pm_aeroequal_files.append(os.path.join(root, file).replace("\\","/"))

            if file.startswith("no2_aeroequal_data.csv"):
                no2_aeroequal_files.append(os.path.join(root, file).replace("\\","/"))

            if file.startswith("iAnemometer.xls"):
                anemometer_files.append(os.path.join(root, file).replace("\\","/"))

    return pm_sensor_files, no2_sensor_files, pm_aeroequal_files, no2_aeroequal_files, anemometer_files

##### 1. Sort the files in increasing order of date
##### 2. The Anemometer files are in Excel, while the others are in CSV.
##### 3. The logic merges a particular sensor file for all dates into one master file.

In [3]:
def create_master_table(files):

    files.sort(key=lambda x: int(x.split('/')[-2].split('_')[1]))
    
    data_frames = []

    for f in files:
        
        if f.endswith('.csv'):
            df = pd.read_csv(f)
            df.columns = df.columns.str.strip()
            if 'Date Time' in df.columns:
                df[['Date', 'Time']] = df['Date Time'].apply(lambda x: pd.Series([parser.parse(x).strftime('%d-%m-%Y')
                                                                                  , parser.parse(x).strftime('%H:%M:%S')]))
                df.drop(columns=['Date Time'], inplace=True)
            else: 
                df['Date'] = df['Date'].apply(lambda x: parser.parse(x.replace(":","/")).strftime('%d-%m-%Y'))

        elif f.endswith('.xls'):
            df = pd.read_excel(f, header=1)
            date_str = f.split('/')[-2].replace('_', '-')
            df['Date'] = parser.parse(date_str).strftime('%d-%m-%Y')
            df.drop(columns=['ID'], inplace=True, errors='ignore')

        df['Time'] = df['Time'].apply(lambda x: parser.parse(x).strftime('%H:%M:%S'))
        df = df[['Date', 'Time'] + [col for col in df.columns if col not in ['Date', 'Time']]]
        data_frames.append(df)

    return pd.concat(data_frames, ignore_index=True)

In [4]:
directory_path = 'E:/Final Dissertation/'
pm_sensor_files, no2_sensor_files, pm_aeroequal_files, no2_aeroequal_files, anemometer_files  = find_csv_files(directory_path)

In [5]:
for item in pm_sensor_files:
    print("PM Sensor File:", item)

PM Sensor File: E:/Final Dissertation/Collected_Data/August_15_2024/dataset.csv
PM Sensor File: E:/Final Dissertation/Collected_Data/August_16_2024/dataset.csv
PM Sensor File: E:/Final Dissertation/Collected_Data/August_18_2024/dataset.csv
PM Sensor File: E:/Final Dissertation/Collected_Data/August_19_2024/dataset.csv
PM Sensor File: E:/Final Dissertation/Collected_Data/August_20_2024/dataset.csv
PM Sensor File: E:/Final Dissertation/Collected_Data/August_21_2024/dataset.csv
PM Sensor File: E:/Final Dissertation/Collected_Data/August_22_2024/dataset.csv


In [6]:
for item in no2_sensor_files:
    print("NO2 Sensor File:", item)

NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_15_2024/no2dataset.csv
NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_16_2024/no2dataset.csv
NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_18_2024/no2dataset.csv
NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_19_2024/no2dataset.csv
NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_20_2024/no2dataset.csv
NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_21_2024/no2dataset.csv
NO2 Sensor File: E:/Final Dissertation/Collected_Data/August_22_2024/no2dataset.csv


In [7]:
for item in pm_aeroequal_files:
    print("PM Aeroequal File:", item)

PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_15_2024/pm_aeroequal_data.csv
PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_16_2024/pm_aeroequal_data.csv
PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_18_2024/pm_aeroequal_data.csv
PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_19_2024/pm_aeroequal_data.csv
PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_20_2024/pm_aeroequal_data.csv
PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_21_2024/pm_aeroequal_data.csv
PM Aeroequal File: E:/Final Dissertation/Collected_Data/August_22_2024/pm_aeroequal_data.csv


In [8]:
for item in no2_aeroequal_files:
    print("NO2 Aeroequal File:", item)

NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_15_2024/no2_aeroequal_data.csv
NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_16_2024/no2_aeroequal_data.csv
NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_18_2024/no2_aeroequal_data.csv
NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_19_2024/no2_aeroequal_data.csv
NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_20_2024/no2_aeroequal_data.csv
NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_21_2024/no2_aeroequal_data.csv
NO2 Aeroequal File: E:/Final Dissertation/Collected_Data/August_22_2024/no2_aeroequal_data.csv


In [9]:
for item in anemometer_files:
    print("Anemometer File:", item)

Anemometer File: E:/Final Dissertation/Collected_Data/August_15_2024/iAnemometer.xls
Anemometer File: E:/Final Dissertation/Collected_Data/August_16_2024/iAnemometer.xls
Anemometer File: E:/Final Dissertation/Collected_Data/August_18_2024/iAnemometer.xls
Anemometer File: E:/Final Dissertation/Collected_Data/August_19_2024/iAnemometer.xls
Anemometer File: E:/Final Dissertation/Collected_Data/August_20_2024/iAnemometer.xls
Anemometer File: E:/Final Dissertation/Collected_Data/August_21_2024/iAnemometer.xls
Anemometer File: E:/Final Dissertation/Collected_Data/August_22_2024/iAnemometer.xls


In [10]:
pm_sensor_master_table = create_master_table(pm_sensor_files)
no2_sensor_master_table = create_master_table(no2_sensor_files)
pm_aeroequal_master_table = create_master_table(pm_aeroequal_files)
no2_aeroequal_master_table = create_master_table(no2_aeroequal_files)
anemometer_master_table = create_master_table(anemometer_files)

In [11]:
pm_sensor_master_table

Date      Time  PM1.0 ug/m3 (ultrafine particles)  \
0     15-08-2024  14:38:00                                  3   
1     15-08-2024  14:39:00                                  3   
2     15-08-2024  14:40:00                                  3   
3     15-08-2024  14:41:00                                  3   
4     15-08-2024  14:42:00                                  3   
...          ...       ...                                ...   
2962  22-08-2024  19:11:09                                  4   
2963  22-08-2024  19:12:09                                  4   
2964  22-08-2024  19:13:09                                  3   
2965  22-08-2024  19:14:09                                  3   
2966  22-08-2024  19:15:09                                  3   

      PM2.5 ug/m3 (combustion particles, organic compounds, metals)  \
0                                                     4               
1                                                     4               
2                                                     4               
3                                                     4               
4                                                     4               
...                                                 ...               
2962                                                 11               
2963                                                 12               
2964                                                 11               
2965                                                 11               
2966                                                 10               

      PM10 ug/m3  (dust, pollen, mould spores)  PM1.0 ug/m3 (atmos env)  \
0                                            4                        3   
1                                            4                        3   
2                                            4                        3   
3                                            4                        3   
4                                            6                        3   
...                                        ...                      ...   
2962                                        13                        4   
2963                                        13                        4   
2964                                        12                        3   
2965                                        12                        3   
2966                                        13                        3   

      PM2.5 ug/m3 (atmos env)  PM10 ug/m3 (atmos env)  >0.3um in 0.1L air  \
0                           4                       4                 516   
1                           4                       4                 546   
2                           4                       4                 567   
3                           4                       4                 597   
4                           4                       6                 597   
...                       ...                     ...                 ...   
2962                       11                      13                 975   
2963                       12                      13                1017   
2964                       11                      12                1017   
2965                       11                      12                1017   
2966                       10                      13                1017   

      >0.5um in 0.1L air  >1.0um in 0.1L air  >2.5um in 0.1L air  \
0                    167                  30                   0   
1                    174                  34                   0   
2                    179                  36                   0   
3                    184                  40                   0   
4                    184                  40                   2   
...                  ...                 ...                 ...   
2962                 281                  67                  20   
2963    

In [12]:
no2_sensor_master_table

Date      Time   NO2 ppb  NO2 µg/m³
0     15-08-2024  14:33:05  0.216462   0.413983
1     15-08-2024  14:34:11 -0.239692  -0.458411
2     15-08-2024  14:35:16  0.071096   0.135970
3     15-08-2024  14:36:21  0.972681   1.860253
4     15-08-2024  14:37:27  0.096102   0.183795
...          ...       ...       ...        ...
2631  22-08-2024  19:09:58  1.146366   2.192424
2632  22-08-2024  19:11:03  1.287635   2.462603
2633  22-08-2024  19:12:09  1.029305   1.968546
2634  22-08-2024  19:13:14  1.104797   2.112924
2635  22-08-2024  19:14:20  1.237881   2.367448

[2636 rows x 4 columns]

In [13]:
pm_aeroequal_master_table

Date      Time  Monitor ID  Location ID  PM10(ppm)  PM2.5(ppm)
0     15-08-2024  14:05:00           1            1      0.001       0.001
1     15-08-2024  14:06:00           1            1      0.001       0.001
2     15-08-2024  14:07:00           1            1      0.000       0.000
3     15-08-2024  14:08:00           1            1      0.001       0.001
4     15-08-2024  14:09:00           1            1      0.001       0.001
...          ...       ...         ...          ...        ...         ...
3042  22-08-2024  19:12:00           1            1      0.000       0.000
3043  22-08-2024  19:13:00           1            1      0.004       0.001
3044  22-08-2024  19:14:00           1            1      0.002       0.001
3045  22-08-2024  19:15:00           1            1      0.002       0.001
3046  22-08-2024  19:16:00           1            1      0.000       0.000

[3047 rows x 6 columns]

In [14]:
no2_aeroequal_master_table

Date      Time  Monitor ID  Location ID  NO2(ppm)
0     15-08-2024  14:05:00           2            1     0.000
1     15-08-2024  14:06:00           2            1     0.000
2     15-08-2024  14:07:00           2            1     0.000
3     15-08-2024  14:08:00           2            1     0.000
4     15-08-2024  14:09:00           2            1     0.000
...          ...       ...         ...          ...       ...
3028  22-08-2024  19:12:00           2            1     0.004
3029  22-08-2024  19:13:00           2            1     0.006
3030  22-08-2024  19:14:00           2            1     0.005
3031  22-08-2024  19:15:00           2            1     0.006
3032  22-08-2024  19:16:00           2            1     0.004

[3033 rows x 5 columns]

In [15]:
anemometer_master_table

Date      Time  Wind Value(km/h)  Temp Value(°C)
0     15-08-2024  15:13:29               0.0            25.6
1     15-08-2024  15:14:29               0.0            26.0
2     15-08-2024  15:15:29               0.0            25.3
3     15-08-2024  15:16:29               0.0            25.4
4     15-08-2024  15:17:29               1.8            25.4
...          ...       ...               ...             ...
2718  22-08-2024  19:12:07               0.3            21.9
2719  22-08-2024  19:13:07               0.0            21.9
2720  22-08-2024  19:14:07               0.0            21.9
2721  22-08-2024  19:15:07               0.0            21.8
2722  22-08-2024  19:16:07               0.0            21.8

[2723 rows x 4 columns]

#### Saved as Master files in output directory in Master_Data.

In [16]:
master_location = "E:/Final Dissertation/Master_Data/"

pm_sensor_master_table.to_csv(master_location + "pm_sensor_master_table.csv", index=False)
no2_sensor_master_table.to_csv(master_location + "no2_sensor_master_table.csv", index=False)
pm_aeroequal_master_table.to_csv(master_location + "pm_aeroequal_master_table.csv", index=False)
no2_aeroequal_master_table.to_csv(master_location + "no2_aeroequal_master_table.csv", index=False)
anemometer_master_table.to_csv(master_location + "anemometer_master_table.csv", index=False)

#### To easily identify whether data was collected by sensor, aeroequal or council as well as reorder columns

In [17]:
column_mapping = {
    'PM2.5 ug/m3 (combustion particles, organic compounds, metals)': 'PM2.5_µg/m³_combustion_particles_organic_compounds_metals_sensor',
    'PM2.5(ppm)': 'PM2.5_mg/m³_aeroequal',
    'PM25': 'PM2.5_µg/m³_council',
    'PM10 ug/m3  (dust, pollen, mould spores)': 'PM10_µg/m³_dust_pollen_mould_spores_sensor',
    'PM10(ppm)': 'PM10_mg/m³_aeroequal',
    'PM10': 'PM10_µg/m³_council',
    'NO2 ppb': 'NO2_ppb_sensor',
    'NO2(ppm)': 'NO2_ppm_aeroequal',
    'NO2': 'NO2_µg/m³_council',
    'Wind Value(km/h)': 'Wind_value_km/h',
    'PM1.0 ug/m3 (ultrafine particles)': 'PM1.0_µg/m³_ultrafine_particles_sensor',
    'PM1.0 ug/m3 (atmos env)': 'PM1.0_µg/m³_atmos_env',
    'PM2.5 ug/m3 (atmos env)': 'PM2.5_µg/m³_atmos_env',
    'PM10 ug/m3 (atmos env)': 'PM10_µg/m³_atmos_env',
    '>0.3um in 0.1L air': 'GT_0.3um_in_0.1l_air',
    '>0.5um in 0.1L air': 'GT_0.5um_in_0.1l_air',
    '>1.0um in 0.1L air': 'GT_1.0um_in_0.1l_air',
    '>2.5um in 0.1L air': 'GT_2.5um_in_0.1l_air',
    '>5.0um in 0.1L air': 'GT_5.0um_in_0.1l_air',
    '>10um in 0.1L air': 'GT_10um_in_0.1l_air',
    'CO2 in ppm': 'CO2_in_ppm',
    'Temperature in C': 'Temperature_in_c',
    'Relative Humidity': 'Relative_humidity',
    'NO2 µg/m³': 'NO2_µg/m³_sensor',
    'Temp Value(°C)': 'Temp_value_°c',
    'NO': 'NO_µg/m³_council',
    'NOXasNO2': 'NOx_as_NO2_µg/m³_council',
    'NV10': 'NV10_µg/m³_council',
    'NV25': 'NV2.5_µg/m³_council',
    'V10': 'V10_µg/m³_council',
    'V25': 'V2.5_µg/m³_council'
}

final_columns = ['Date', 'Time'] + list(column_mapping.values())

In [18]:
def load_and_preprocess(file):
    df = pd.read_csv(file).rename(columns=str.strip)
    df['DateTime'] = pd.to_datetime(df.pop('Date') + ' ' + df.pop('Time'), format='%d-%m-%Y %H:%M:%S')
    return df.sort_values('DateTime')

#### The master files are merged with a tolerance of 29 seconds, choosing the nearest available value
#### The DateTime is selected from the file with the second lowest number of records.

In [19]:
def perform_merge_sensor_aeroequal(tolerance_seconds, master_files):
    dfs = [load_and_preprocess(file) for file in master_files]
    tolerance = pd.Timedelta(seconds=tolerance_seconds)
    
    merged_df = dfs[0]
    for df in dfs[1:]:
        merged_df = pd.merge_asof(merged_df, df, on='DateTime', tolerance=tolerance, direction='nearest')
    
    merged_df.drop(columns=[col for col in merged_df if 'Location ID' in col or 'Monitor ID' in col], inplace=True)
    
    merged_df['Date'] = merged_df['DateTime'].dt.date
    merged_df['Time'] = merged_df['DateTime'].dt.time
    merged_df.drop(columns=['DateTime'], inplace=True)
    
    primary_columns = [
        'Date', 'Time',
        'PM2.5 ug/m3 (combustion particles, organic compounds, metals)',  
        'PM2.5(ppm)',  
        'PM10 ug/m3  (dust, pollen, mould spores)',  
        'PM10(ppm)',  
        'NO2 ppb',  
        'NO2(ppm)',
        'Wind Value(km/h)'
    ]
    
    final_df = merged_df[primary_columns + [col for col in merged_df if col not in primary_columns]]
    final_df = final_df.apply(lambda col: col.interpolate(method='linear') if col.dtype.kind in 'biufc' else col)
    final_df.rename(columns=column_mapping, inplace=True)
    
    return final_df

In [20]:
directory_path = 'E:/Final Dissertation/Master_Data/'
master_files = [os.path.join(directory_path, file) for file in [
    'anemometer_master_table.csv',
    'no2_aeroequal_master_table.csv',
    'no2_sensor_master_table.csv',
    'pm_aeroequal_master_table.csv',
    'pm_sensor_master_table.csv'
]]

#### Why tolerance = 29?
#### 1. Tolerance of less than 29 generated NaN cells with 2723 records, meaning the nearest value was unavailable.
#### 2. Tolerance greater than 29 had no impact.

In [21]:
tolerance = 29
final_df = perform_merge_sensor_aeroequal(tolerance, master_files)
final_df.to_csv('E:/Final Dissertation/Final_Dataset/Final_Dataset.csv', index=False)

#### This marks the end of generating a dataset sampled every minute.
#### Although not utilized for the current problem statement, this Dataset will be helpful for future research.

#### This marks the beginning of generating a dataset sampled every hour, matching the council dataset.
#### The Date Time column from Council is taken as reference, using the same logic as above but with the tolerance of 1 hour

In [22]:
council_df = pd.read_csv('E:/Final Dissertation/Final_Dataset/2024-08-14-240826181306.csv')
council_df['End Date'] = pd.to_datetime(council_df['End Date'].ffill(), format='%d-%m-%Y', errors='coerce')
council_df['End Time'] = council_df['End Time'].replace('24:00:00', '00:00:00')
council_df.loc[council_df['End Time'] == '00:00:00', 'End Date'] += pd.DateOffset(days=1)
council_df['DateTime'] = pd.to_datetime(council_df['End Date'].dt.strftime('%d-%m-%Y') + ' ' + council_df['End Time'], format='%d-%m-%Y %H:%M:%S')
numeric_cols = council_df.select_dtypes(include='number').columns
council_df = council_df.set_index('DateTime')[numeric_cols].resample('h').mean().reset_index()

In [23]:
def preprocess_master_df(file):
    df = pd.read_csv(file)
    df['DateTime'] = pd.to_datetime(df.pop('Date') + ' ' + df.pop('Time'), format='%d-%m-%Y %H:%M:%S')
    return df.set_index('DateTime').resample('h').mean().reset_index()

In [24]:
def perform_merge_sensor_aeroequal_council():

    anemometer_df = preprocess_master_df('E:/Final Dissertation/Master_Data/anemometer_master_table.csv')
    no2_aeroequal_df = preprocess_master_df('E:/Final Dissertation/Master_Data/no2_aeroequal_master_table.csv')
    no2_sensor_df = preprocess_master_df('E:/Final Dissertation/Master_Data/no2_sensor_master_table.csv')
    pm_aeroequal_df = preprocess_master_df('E:/Final Dissertation/Master_Data/pm_aeroequal_master_table.csv')
    pm_sensor_df = preprocess_master_df('E:/Final Dissertation/Master_Data/pm_sensor_master_table.csv')
    
    merged_df = anemometer_df
    for df in [no2_aeroequal_df, no2_sensor_df, pm_aeroequal_df, pm_sensor_df, council_df]:
        merged_df = pd.merge_asof(merged_df, df, on='DateTime', tolerance=pd.Timedelta(hours=1), direction='nearest')
    
    merged_df.rename(columns=column_mapping, inplace=True)
    merged_df = merged_df.dropna(subset=[col for col in merged_df.columns if '_council' not in col], how='all')
    
    merged_df['Date'] = merged_df['DateTime'].dt.date
    merged_df['Time'] = merged_df['DateTime'].dt.time
    merged_df.drop(columns=['DateTime'], inplace=True)

    return merged_df[final_columns].dropna()

In [25]:
final_df = perform_merge_sensor_aeroequal_council()
final_df.to_csv('E:/Final Dissertation/Final_Dataset/Final_Hourly_Dataset.csv', index=False)

In [26]:
expected_columns = list(column_mapping.values())

In [27]:
def generate_statistics(final_df, master_files):
    
    for col in expected_columns:
        if col not in final_df.columns:
            final_df[col] = pd.NA
    
    present_columns = [col for col in expected_columns if col in final_df.columns]
    
    final_stats = final_df[present_columns].describe().T[['count', 'mean', 'std', 'min', '25%', '50%', '75%', 'max']]

    master_stats_combined = pd.DataFrame()
    for file_path in master_files:
        df = pd.read_csv(file_path)
        df.columns = [col.strip() for col in df.columns]
        
        df.rename(columns=column_mapping, inplace=True)
        
        common_columns = [col for col in present_columns if col in df.columns]
        if common_columns:
            stats = df[common_columns].describe().T[['count', 'mean', 'std', 'min', '25%', '50%', '75%', 'max']]
            master_stats_combined = pd.concat([master_stats_combined, stats])

    master_stats_combined = master_stats_combined.reindex(present_columns)
    
    return final_stats, master_stats_combined

In [28]:
directory_path = 'E:/Final Dissertation/Master_Data/'

master_files = [os.path.join(directory_path, file) for file in [
    'anemometer_master_table.csv',
    'no2_aeroequal_master_table.csv',
    'no2_sensor_master_table.csv',
    'pm_aeroequal_master_table.csv',
    'pm_sensor_master_table.csv'
]]

final_df = pd.read_csv('E:/Final Dissertation/Final_Dataset/Final_Dataset.csv')
final_hourly_df = pd.read_csv('E:/Final Dissertation/Final_Dataset/Final_Hourly_Dataset.csv')
final_stats, master_stats_combined = generate_statistics(final_df, master_files)
final_hourly_stats, _ = generate_statistics(final_hourly_df, master_files)

#### This marks the end of generating a dataset sampled every hour.

##### 1. Statistics to see how the merged files are different from the master files combined.
##### 2. NaN, in this case, is not available in the file
##### 3. Council columns are not present in the master file and Final Dataset sampled every minute
##### 4. Some Sensor Columns are not present in the council file

In [29]:
comparison_count = pd.DataFrame({
    'Final Dataset Count': final_stats['count'],
    'Final Hourly Dataset Count': final_hourly_stats['count'],
    'Master Files Combined Count': master_stats_combined['count']
})
comparison_count.reindex(expected_columns)

Final Dataset Count  \
PM2.5_µg/m³_combustion_particles_organic_compou...               2723.0   
PM2.5_mg/m³_aeroequal                                            2723.0   
PM2.5_µg/m³_council                                                 NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                       2723.0   
PM10_mg/m³_aeroequal                                             2723.0   
PM10_µg/m³_council                                                  NaN   
NO2_ppb_sensor                                                   2723.0   
NO2_ppm_aeroequal                                                2723.0   
NO2_µg/m³_council                                                   NaN   
Wind_value_km/h                                                  2723.0   
PM1.0_µg/m³_ultrafine_particles_sensor                           2723.0   
PM1.0_µg/m³_atmos_env                                            2723.0   
PM2.5_µg/m³_atmos_env                                            2723.0   
PM10_µg/m³_atmos_env                                             2723.0   
GT_0.3um_in_0.1l_air                                             2723.0   
GT_0.5um_in_0.1l_air                                             2723.0   
GT_1.0um_in_0.1l_air                                             2723.0   
GT_2.5um_in_0.1l_air                                             2723.0   
GT_5.0um_in_0.1l_air                                             2723.0   
GT_10um_in_0.1l_air                                              2723.0   
CO2_in_ppm                                                       2723.0   
Temperature_in_c                                                 2723.0   
Relative_humidity                                                2723.0   
NO2_µg/m³_sensor                                                 2723.0   
Temp_value_°c                                                    2723.0   
NO_µg/m³_council                                                    NaN   
NOx_as_NO2_µg/m³_council                                            NaN   
NV10_µg/m³_council                                                  NaN   
NV2.5_µg/m³_council                                                 NaN   
V10_µg/m³_council                                                   NaN   
V2.5_µg/m³_council                                                  NaN   

                                                    Final Hourly Dataset Count  \
PM2.5_µg/m³_combustion_particles_organic_compou...                        55.0   
PM2.5_mg/m³_aeroequal                                                     55.0   
PM2.5_µg/m³_council                                                       55.0   
PM10_µg/m³_dust_pollen_mould_spores_sensor                                55.0   
PM10_mg/m³_aeroequal                                                      55.0   
PM10_µg/m³_council                                                        55.0   
NO2_ppb_sensor                                                            55.0   
NO2_ppm_aeroequal                                                         55.0   
NO2_µg/m³_council                                                         55.0   
Wind_value_km/h                                                           55.0   
PM1.0_µg/m³_ultrafine_particles_sensor                                    55.0   
PM1.0_µg/m³_atmos_env                                                     55.0   
PM2.5_µg/m³_atmos_env                                                     55.0   
PM10_µg/m³_atmos_env                                                      55.0   
GT_0.3um_in_0.1l_air                                                      55.0   
GT_0.5um_in_0.1l_air                                                      55.0   
GT_1.0um_in_0.1l_air                                                      55.0   
GT_2.5um_in_0.1l_air                                                      55.0   
GT_5.0um_in_0.1l_air                                                      55.0   
GT_10um_in_

In [30]:
comparison_mean = pd.DataFrame({
    'Final Dataset Mean': final_stats['mean'],
    'Final Hourly Dataset Mean': final_hourly_stats['mean'],
    'Master Files Combined Mean': master_stats_combined['mean']
})
comparison_mean.reindex(expected_columns)

Final Dataset Mean  \
PM2.5_µg/m³_combustion_particles_organic_compou...           13.278186   
PM2.5_mg/m³_aeroequal                                         0.001366   
PM2.5_µg/m³_council                                                NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                   15.506243   
PM10_mg/m³_aeroequal                                          0.003429   
PM10_µg/m³_council                                                 NaN   
NO2_ppb_sensor                                               -0.400421   
NO2_ppm_aeroequal                                             0.001492   
NO2_µg/m³_council                                                  NaN   
Wind_value_km/h                                               0.269152   
PM1.0_µg/m³_ultrafine_particles_sensor                        7.139736   
PM1.0_µg/m³_atmos_env                                         6.643592   
PM2.5_µg/m³_atmos_env                                        11.702534   
PM10_µg/m³_atmos_env                                         14.532501   
GT_0.3um_in_0.1l_air                                       1307.525707   
GT_0.5um_in_0.1l_air                                        390.327580   
GT_1.0um_in_0.1l_air                                        102.358245   
GT_2.5um_in_0.1l_air                                         12.779471   
GT_5.0um_in_0.1l_air                                          2.840250   
GT_10um_in_0.1l_air                                           2.221447   
CO2_in_ppm                                                  417.997506   
Temperature_in_c                                             24.991647   
Relative_humidity                                            51.418491   
NO2_µg/m³_sensor                                             -0.765806   
Temp_value_°c                                                23.297943   
NO_µg/m³_council                                                   NaN   
NOx_as_NO2_µg/m³_council                                           NaN   
NV10_µg/m³_council                                                 NaN   
NV2.5_µg/m³_council                                                NaN   
V10_µg/m³_council                                                  NaN   
V2.5_µg/m³_council                                                 NaN   

                                                    Final Hourly Dataset Mean  \
PM2.5_µg/m³_combustion_particles_organic_compou...                  12.738731   
PM2.5_mg/m³_aeroequal                                                0.001292   
PM2.5_µg/m³_council                                                  8.789091   
PM10_µg/m³_dust_pollen_mould_spores_sensor                          14.935854   
PM10_mg/m³_aeroequal                                                 0.003314   
PM10_µg/m³_council                                                  17.118182   
NO2_ppb_sensor                                                      -0.485830   
NO2_ppm_aeroequal                                                    0.001950   
NO2_µg/m³_council                                                   14.023581   
Wind_value_km/h                                                      0.275688   
PM1.0_µg/m³_ultrafine_particles_sensor                               6.795935   
PM1.0_µg/m³_atmos_env                                                6.332602   
PM2.5_µg/m³_atmos_env                                               11.244378   
PM10_µg/m³_atmos_env                                                14.001860   
GT_0.3um_in_0.1l_air                                              1258.186158   
GT_0.5um_in_0.1l_air                                               375.992209   
GT_1.0um_in_0.1l_air                                                98.397636   
GT_2.5um_in_0.1l_air                                                12.451079   
GT_5.0um_in_0.1l_air                                                 2.831479   
GT_10um_in_0.1l_air                                            

In [31]:
comparison_std = pd.DataFrame({
    'Final Dataset Std Dev': final_stats['std'],
    'Final Hourly Dataset Std Dev': final_hourly_stats['std'],
    'Master Files Combined Std Dev': master_stats_combined['std']
}).reindex(expected_columns)
comparison_std.reindex(expected_columns)

Final Dataset Std Dev  \
PM2.5_µg/m³_combustion_particles_organic_compou...              17.375500   
PM2.5_mg/m³_aeroequal                                            0.001564   
PM2.5_µg/m³_council                                                   NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                      19.040585   
PM10_mg/m³_aeroequal                                             0.005312   
PM10_µg/m³_council                                                    NaN   
NO2_ppb_sensor                                                   2.427300   
NO2_ppm_aeroequal                                                0.002703   
NO2_µg/m³_council                                                     NaN   
Wind_value_km/h                                                  0.576094   
PM1.0_µg/m³_ultrafine_particles_sensor                           8.753120   
PM1.0_µg/m³_atmos_env                                            7.446158   
PM2.5_µg/m³_atmos_env                                           13.181124   
PM10_µg/m³_atmos_env                                            16.406246   
GT_0.3um_in_0.1l_air                                          1423.979490   
GT_0.5um_in_0.1l_air                                           428.503022   
GT_1.0um_in_0.1l_air                                           135.904285   
GT_2.5um_in_0.1l_air                                            16.856321   
GT_5.0um_in_0.1l_air                                             3.084841   
GT_10um_in_0.1l_air                                              2.673218   
CO2_in_ppm                                                      69.351263   
Temperature_in_c                                                 1.389032   
Relative_humidity                                                6.825561   
NO2_µg/m³_sensor                                                 4.642212   
Temp_value_°c                                                    1.656689   
NO_µg/m³_council                                                      NaN   
NOx_as_NO2_µg/m³_council                                              NaN   
NV10_µg/m³_council                                                    NaN   
NV2.5_µg/m³_council                                                   NaN   
V10_µg/m³_council                                                     NaN   
V2.5_µg/m³_council                                                    NaN   

                                                    Final Hourly Dataset Std Dev  \
PM2.5_µg/m³_combustion_particles_organic_compou...                     16.939382   
PM2.5_mg/m³_aeroequal                                                   0.001386   
PM2.5_µg/m³_council                                                     7.651450   
PM10_µg/m³_dust_pollen_mould_spores_sensor                             18.538678   
PM10_mg/m³_aeroequal                                                    0.002926   
PM10_µg/m³_council                                                      9.930725   
NO2_ppb_sensor                                                          1.860480   
NO2_ppm_aeroequal                                                       0.003165   
NO2_µg/m³_council                                                       4.581423   
Wind_value_km/h                                                         0.263145   
PM1.0_µg/m³_ultrafine_particles_sensor                                  8.488992   
PM1.0_µg/m³_atmos_env                                                   7.235916   
PM2.5_µg/m³_atmos_env                                                  12.842817   
PM10_µg/m³_atmos_env                                                   15.949654   
GT_0.3um_in_0.1l_air                                                 1390.457257   
GT_0.5um_in_0.1l_air                                                  418.012283   
GT_1.0um_in_0.1l_air                                                  132.374126   
GT_2.5um_in_0.1l_air                                                   16.1

In [32]:
comparison_min = pd.DataFrame({
    'Final Dataset Min': final_stats['min'],
    'Final Hourly Dataset Min': final_hourly_stats['min'],
    'Master Files Combined Min': master_stats_combined['min']
})
comparison_min.reindex(expected_columns)

Final Dataset Min  \
PM2.5_µg/m³_combustion_particles_organic_compou...           0.000000   
PM2.5_mg/m³_aeroequal                                        0.000000   
PM2.5_µg/m³_council                                               NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                   0.000000   
PM10_mg/m³_aeroequal                                         0.000000   
PM10_µg/m³_council                                                NaN   
NO2_ppb_sensor                                             -63.952923   
NO2_ppm_aeroequal                                            0.000000   
NO2_µg/m³_council                                                 NaN   
Wind_value_km/h                                              0.000000   
PM1.0_µg/m³_ultrafine_particles_sensor                       0.000000   
PM1.0_µg/m³_atmos_env                                        0.000000   
PM2.5_µg/m³_atmos_env                                        0.000000   
PM10_µg/m³_atmos_env                                         0.000000   
GT_0.3um_in_0.1l_air                                       135.000000   
GT_0.5um_in_0.1l_air                                        37.000000   
GT_1.0um_in_0.1l_air                                         4.000000   
GT_2.5um_in_0.1l_air                                         0.000000   
GT_5.0um_in_0.1l_air                                         0.000000   
GT_10um_in_0.1l_air                                          0.000000   
CO2_in_ppm                                                 208.270000   
Temperature_in_c                                            22.470000   
Relative_humidity                                           38.990000   
NO2_µg/m³_sensor                                          -122.309959   
Temp_value_°c                                               19.700000   
NO_µg/m³_council                                                  NaN   
NOx_as_NO2_µg/m³_council                                          NaN   
NV10_µg/m³_council                                                NaN   
NV2.5_µg/m³_council                                               NaN   
V10_µg/m³_council                                                 NaN   
V2.5_µg/m³_council                                                NaN   

                                                    Final Hourly Dataset Min  \
PM2.5_µg/m³_combustion_particles_organic_compou...                  0.750000   
PM2.5_mg/m³_aeroequal                                               0.000083   
PM2.5_µg/m³_council                                                -1.000000   
PM10_µg/m³_dust_pollen_mould_spores_sensor                          1.600000   
PM10_mg/m³_aeroequal                                                0.000485   
PM10_µg/m³_council                                                  0.000000   
NO2_ppb_sensor                                                     -5.703395   
NO2_ppm_aeroequal                                                   0.000000   
NO2_µg/m³_council                                                   7.975130   
Wind_value_km/h                                                     0.000000   
PM1.0_µg/m³_ultrafine_particles_sensor                              0.000000   
PM1.0_µg/m³_atmos_env                                               0.000000   
PM2.5_µg/m³_atmos_env                                               0.750000   
PM10_µg/m³_atmos_env                                                1.600000   
GT_0.3um_in_0.1l_air                                              193.700000   
GT_0.5um_in_0.1l_air                                               52.150000   
GT_1.0um_in_0.1l_air                                               10.800000   
GT_2.5um_in_0.1l_air                                                1.000000   
GT_5.0um_in_0.1l_air                                                0.000000   
GT_10um_in_0.1l_air                                                 0.000000   
CO2_in_ppm                         

In [33]:
comparison_25 = pd.DataFrame({
    'Final Dataset 25%': final_stats['25%'],
    'Final Hourly Dataset 25%': final_hourly_stats['25%'],
    'Master Files Combined 25%': master_stats_combined['25%']
})
comparison_25.reindex(expected_columns)

Final Dataset 25%  \
PM2.5_µg/m³_combustion_particles_organic_compou...           4.000000   
PM2.5_mg/m³_aeroequal                                        0.000000   
PM2.5_µg/m³_council                                               NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                   5.000000   
PM10_mg/m³_aeroequal                                         0.000000   
PM10_µg/m³_council                                                NaN   
NO2_ppb_sensor                                              -1.607338   
NO2_ppm_aeroequal                                            0.000000   
NO2_µg/m³_council                                                 NaN   
Wind_value_km/h                                              0.000000   
PM1.0_µg/m³_ultrafine_particles_sensor                       3.000000   
PM1.0_µg/m³_atmos_env                                        3.000000   
PM2.5_µg/m³_atmos_env                                        4.000000   
PM10_µg/m³_atmos_env                                         5.000000   
GT_0.3um_in_0.1l_air                                       570.000000   
GT_0.5um_in_0.1l_air                                       169.000000   
GT_1.0um_in_0.1l_air                                        31.000000   
GT_2.5um_in_0.1l_air                                         2.000000   
GT_5.0um_in_0.1l_air                                         0.000000   
GT_10um_in_0.1l_air                                          0.000000   
CO2_in_ppm                                                 381.500000   
Temperature_in_c                                            23.930000   
Relative_humidity                                           45.410000   
NO2_µg/m³_sensor                                            -3.074033   
Temp_value_°c                                               22.200000   
NO_µg/m³_council                                                  NaN   
NOx_as_NO2_µg/m³_council                                          NaN   
NV10_µg/m³_council                                                NaN   
NV2.5_µg/m³_council                                               NaN   
V10_µg/m³_council                                                 NaN   
V2.5_µg/m³_council                                                NaN   

                                                    Final Hourly Dataset 25%  \
PM2.5_µg/m³_combustion_particles_organic_compou...                  4.608333   
PM2.5_mg/m³_aeroequal                                               0.000442   
PM2.5_µg/m³_council                                                 4.150000   
PM10_µg/m³_dust_pollen_mould_spores_sensor                          5.798023   
PM10_mg/m³_aeroequal                                                0.001255   
PM10_µg/m³_council                                                 10.800000   
NO2_ppb_sensor                                                     -1.661391   
NO2_ppm_aeroequal                                                   0.000000   
NO2_µg/m³_council                                                  10.870170   
Wind_value_km/h                                                     0.081667   
PM1.0_µg/m³_ultrafine_particles_sensor                              2.710417   
PM1.0_µg/m³_atmos_env                                               2.710417   
PM2.5_µg/m³_atmos_env                                               4.608333   
PM10_µg/m³_atmos_env                                                5.798023   
GT_0.3um_in_0.1l_air                                              622.025000   
GT_0.5um_in_0.1l_air                                              186.712429   
GT_1.0um_in_0.1l_air                                               32.800000   
GT_2.5um_in_0.1l_air                                                3.850000   
GT_5.0um_in_0.1l_air                                                1.053333   
GT_10um_in_0.1l_air                                                 0.966667   
CO2_in_ppm                         

In [34]:
comparison_50 = pd.DataFrame({
    'Final Dataset 50%': final_stats['50%'],
    'Final Hourly Dataset 50%': final_hourly_stats['50%'],
    'Master Files Combined 50%': master_stats_combined['50%']
})
comparison_50.reindex(expected_columns)

Final Dataset 50%  \
PM2.5_µg/m³_combustion_particles_organic_compou...           7.000000   
PM2.5_mg/m³_aeroequal                                        0.001000   
PM2.5_µg/m³_council                                               NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                   9.000000   
PM10_mg/m³_aeroequal                                         0.001000   
PM10_µg/m³_council                                                NaN   
NO2_ppb_sensor                                               0.022329   
NO2_ppm_aeroequal                                            0.000000   
NO2_µg/m³_council                                                 NaN   
Wind_value_km/h                                              0.000000   
PM1.0_µg/m³_ultrafine_particles_sensor                       4.000000   
PM1.0_µg/m³_atmos_env                                        4.000000   
PM2.5_µg/m³_atmos_env                                        7.000000   
PM10_µg/m³_atmos_env                                         9.000000   
GT_0.3um_in_0.1l_air                                       828.000000   
GT_0.5um_in_0.1l_air                                       243.000000   
GT_1.0um_in_0.1l_air                                        51.000000   
GT_2.5um_in_0.1l_air                                         6.000000   
GT_5.0um_in_0.1l_air                                         2.000000   
GT_10um_in_0.1l_air                                          2.000000   
CO2_in_ppm                                                 409.740000   
Temperature_in_c                                            24.820000   
Relative_humidity                                           49.930000   
NO2_µg/m³_sensor                                             0.042704   
Temp_value_°c                                               22.900000   
NO_µg/m³_council                                                  NaN   
NOx_as_NO2_µg/m³_council                                          NaN   
NV10_µg/m³_council                                                NaN   
NV2.5_µg/m³_council                                               NaN   
V10_µg/m³_council                                                 NaN   
V2.5_µg/m³_council                                                NaN   

                                                    Final Hourly Dataset 50%  \
PM2.5_µg/m³_combustion_particles_organic_compou...                  7.133333   
PM2.5_mg/m³_aeroequal                                               0.000700   
PM2.5_µg/m³_council                                                 5.900000   
PM10_µg/m³_dust_pollen_mould_spores_sensor                          8.250000   
PM10_mg/m³_aeroequal                                                0.002200   
PM10_µg/m³_council                                                 14.100000   
NO2_ppb_sensor                                                     -0.005002   
NO2_ppm_aeroequal                                                   0.000217   
NO2_µg/m³_council                                                  12.761160   
Wind_value_km/h                                                     0.181667   
PM1.0_µg/m³_ultrafine_particles_sensor                              3.800000   
PM1.0_µg/m³_atmos_env                                               3.800000   
PM2.5_µg/m³_atmos_env                                               7.133333   
PM10_µg/m³_atmos_env                                                8.250000   
GT_0.3um_in_0.1l_air                                              810.150000   
GT_0.5um_in_0.1l_air                                              236.800000   
GT_1.0um_in_0.1l_air                                               50.600000   
GT_2.5um_in_0.1l_air                                                6.705882   
GT_5.0um_in_0.1l_air                                                2.000000   
GT_10um_in_0.1l_air                                                 1.533333   
CO2_in_ppm                         

In [35]:
comparison_75 = pd.DataFrame({
    'Final Dataset 75%': final_stats['75%'],
    'Final Hourly Dataset 75%': final_hourly_stats['75%'],
    'Master Files Combined 75%': master_stats_combined['75%']
})
comparison_75.reindex(expected_columns)

Final Dataset 75%  \
PM2.5_µg/m³_combustion_particles_organic_compou...          11.000000   
PM2.5_mg/m³_aeroequal                                        0.002000   
PM2.5_µg/m³_council                                               NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                  14.000000   
PM10_mg/m³_aeroequal                                         0.004000   
PM10_µg/m³_council                                                NaN   
NO2_ppb_sensor                                               0.743057   
NO2_ppm_aeroequal                                            0.002750   
NO2_µg/m³_council                                                 NaN   
Wind_value_km/h                                              0.300000   
PM1.0_µg/m³_ultrafine_particles_sensor                       6.000000   
PM1.0_µg/m³_atmos_env                                        6.000000   
PM2.5_µg/m³_atmos_env                                       11.000000   
PM10_µg/m³_atmos_env                                        14.000000   
GT_0.3um_in_0.1l_air                                      1131.000000   
GT_0.5um_in_0.1l_air                                       341.000000   
GT_1.0um_in_0.1l_air                                        76.000000   
GT_2.5um_in_0.1l_air                                        12.000000   
GT_5.0um_in_0.1l_air                                         4.000000   
GT_10um_in_0.1l_air                                          4.000000   
CO2_in_ppm                                                 438.630000   
Temperature_in_c                                            25.900000   
Relative_humidity                                           56.410000   
NO2_µg/m³_sensor                                             1.421096   
Temp_value_°c                                               24.600000   
NO_µg/m³_council                                                  NaN   
NOx_as_NO2_µg/m³_council                                          NaN   
NV10_µg/m³_council                                                NaN   
NV2.5_µg/m³_council                                               NaN   
V10_µg/m³_council                                                 NaN   
V2.5_µg/m³_council                                                NaN   

                                                    Final Hourly Dataset 75%  \
PM2.5_µg/m³_combustion_particles_organic_compou...                  9.383333   
PM2.5_mg/m³_aeroequal                                               0.001608   
PM2.5_µg/m³_council                                                11.100000   
PM10_µg/m³_dust_pollen_mould_spores_sensor                         11.641667   
PM10_mg/m³_aeroequal                                                0.003850   
PM10_µg/m³_council                                                 22.100000   
NO2_ppb_sensor                                                      0.759264   
NO2_ppm_aeroequal                                                   0.003613   
NO2_µg/m³_council                                                  15.347815   
Wind_value_km/h                                                     0.459167   
PM1.0_µg/m³_ultrafine_particles_sensor                              5.723529   
PM1.0_µg/m³_atmos_env                                               5.723529   
PM2.5_µg/m³_atmos_env                                               9.383333   
PM10_µg/m³_atmos_env                                               11.641667   
GT_0.3um_in_0.1l_air                                             1010.175000   
GT_0.5um_in_0.1l_air                                              302.450000   
GT_1.0um_in_0.1l_air                                               63.731250   
GT_2.5um_in_0.1l_air                                               11.416667   
GT_5.0um_in_0.1l_air                                                4.050000   
GT_10um_in_0.1l_air                                                 3.056250   
CO2_in_ppm                         

In [36]:
comparison_max = pd.DataFrame({
    'Final Dataset Max': final_stats['max'],
    'Final Hourly Dataset Max': final_hourly_stats['max'],
    'Master Files Combined Max': master_stats_combined['max']
})
comparison_max.reindex(expected_columns)

Final Dataset Max  \
PM2.5_µg/m³_combustion_particles_organic_compou...          70.000000   
PM2.5_mg/m³_aeroequal                                        0.008000   
PM2.5_µg/m³_council                                               NaN   
PM10_µg/m³_dust_pollen_mould_spores_sensor                  80.000000   
PM10_mg/m³_aeroequal                                         0.097000   
PM10_µg/m³_council                                                NaN   
NO2_ppb_sensor                                              12.758244   
NO2_ppm_aeroequal                                            0.030000   
NO2_µg/m³_council                                                 NaN   
Wind_value_km/h                                              4.300000   
PM1.0_µg/m³_ultrafine_particles_sensor                      34.000000   
PM1.0_µg/m³_atmos_env                                       28.000000   
PM2.5_µg/m³_atmos_env                                       50.000000   
PM10_µg/m³_atmos_env                                        64.000000   
GT_0.3um_in_0.1l_air                                      5535.000000   
GT_0.5um_in_0.1l_air                                      1663.000000   
GT_1.0um_in_0.1l_air                                       553.000000   
GT_2.5um_in_0.1l_air                                        84.000000   
GT_5.0um_in_0.1l_air                                        20.000000   
GT_10um_in_0.1l_air                                         18.000000   
CO2_in_ppm                                                1669.060000   
Temperature_in_c                                            28.660000   
Relative_humidity                                           66.190000   
NO2_µg/m³_sensor                                            24.400144   
Temp_value_°c                                               30.100000   
NO_µg/m³_council                                                  NaN   
NOx_as_NO2_µg/m³_council                                          NaN   
NV10_µg/m³_council                                                NaN   
NV2.5_µg/m³_council                                               NaN   
V10_µg/m³_council                                                 NaN   
V2.5_µg/m³_council                                                NaN   

                                                    Final Hourly Dataset Max  \
PM2.5_µg/m³_combustion_particles_organic_compou...                 62.800000   
PM2.5_mg/m³_aeroequal                                               0.005183   
PM2.5_µg/m³_council                                                30.000000   
PM10_µg/m³_dust_pollen_mould_spores_sensor                         70.750000   
PM10_mg/m³_aeroequal                                                0.011383   
PM10_µg/m³_council                                                 49.700000   
NO2_ppb_sensor                                                      4.027257   
NO2_ppm_aeroequal                                                   0.016950   
NO2_µg/m³_council                                                  28.128090   
Wind_value_km/h                                                     0.956250   
PM1.0_µg/m³_ultrafine_particles_sensor                             31.500000   
PM1.0_µg/m³_atmos_env                                              26.333333   
PM2.5_µg/m³_atmos_env                                              46.466667   
PM10_µg/m³_atmos_env                                               59.766667   
GT_0.3um_in_0.1l_air                                             5253.100000   
GT_0.5um_in_0.1l_air                                             1575.833333   
GT_1.0um_in_0.1l_air                                              498.900000   
GT_2.5um_in_0.1l_air                                               64.545455   
GT_5.0um_in_0.1l_air                                               10.727273   
GT_10um_in_0.1l_air                                                 9.200000   
CO2_in_ppm                         